src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java

```diff
@@ -18,77 +17,119 @@
 
 public class NavigationBar extends WikiBasePageObject {
 
-    @FindBy(css = "#searchInput")
-    private WebElement searchInput;
-    @FindBy(css = "#searchSubmit")
-    private WebElement searchSubmit;
-    @FindBy(css = ".autocomplete div")
-    private WebElement suggestion;
-    @FindBys(@FindBy(css = ".autocomplete div"))
-    private List<WebElement> suggestionsList;
-
-    private By jqueryAutocompleteBy = By.cssSelector("[src*='jquery.autocomplete']");
-
-    public NavigationBar(WebDriver driver) {
+	public NavigationBar(WebDriver driver) {
         super(driver);
     }
 
-    public void triggerSuggestions(String query) {
-        searchInput.click();
-        waitForElementByBy(jqueryAutocompleteBy);
-        sendKeysHumanSpeed(searchInput, query);
-        waitForElementByElement(suggestion);		
-        waitForElementByElement(suggestionsList.get(0));		
-    }
-
-    public void verifySuggestions(String suggestion) {
-        waitForElementByElement(suggestionsList.get(0));
-        String allSuggestions = "";
-        for (int i = 0; i < suggestionsList.size(); i++) {
-            allSuggestions += suggestionsList.get(i).getAttribute("title");
-        }
-        Assertion.assertStringContains(suggestion, allSuggestions);
-    }
+    @FindBy(css = "#searchInput")
+	private WebElement searchInput;
+	@FindBy(css = "#searchSubmit")
+	private WebElement searchSubmit;
+	@FindBy(css = ".autocomplete div")
+	private WebElement suggestion;
+	@FindBys(@FindBy(css = ".autocomplete div"))
+	private List<WebElement> suggestionsList;
+	private String suggestionCss = ".autocomplete div"; 
+	
+    private By jqueryAutocompleteBy = By.cssSelector("[src*='jquery.autocomplete']");
 
-    /**
-     * This method will arrow down through suggestions 
-     * until it finds given suggestion, then it clicks enter.
-     */
-    public ArticlePageObject ArrowDownAndEnterSuggestion(String suggestion) {
-        waitForElementByElement(suggestionsList.get(0));
-        for (int i = 0; i < suggestionsList.size(); i++) {
-            WebElement currentSuggestion = suggestionsList.get(i);
-            searchInput.sendKeys(Keys.ARROW_DOWN);
-            if (currentSuggestion.getText().contains(suggestion)) {
-                searchInput.sendKeys(Keys.ENTER);
-                PageObjectLogging.log("ArrowDownToSuggestion", 
-                        "arrowed down to desired suggestion" +suggestion+"and clicked enter",
-                        true);
-                return new ArticlePageObject(driver);
-            }
-            else {
-                searchInput.sendKeys(Keys.ARROW_DOWN);
-            }
-        }
-        PageObjectLogging.log("ArrowDownToSuggestion", "didn't find suggestion: "+suggestion, true);
-        return null;
-    }
+	public void triggerSuggestions(String query) {
+		waitForElementByElement(searchInput);
+		searchInput.clear();
+		waitForElementClickableByElement(searchInput);
```

Comment:
yes, but second wait is wait for clickability and after clearing searchInput. this has meaning for stability.